html_css/v2/main-course/custom-properties.md

```diff
@@ -0,0 +1,205 @@
+
+#### The `:root:` Selector
+
+While there may be times where you will want to limit the scope of a custom property, you may want to be able to use other custom properties on many, unrelated selectors. One work around would be declaring the same custom property on a bunch of selectors, but that defeats one of the purposes of using custom properties in the first place (the ease of changing multiple instances of a value at once).
+
+A better solution is declaring those custom properties on the `:root` selector, which is essentially the same thing as the `html` selector except it has a higher specificity.
+
+~~~html
+<p class='cool-paragraph'>Lorem ipsum dolor sit amet.</p>
+
+<p class='exciting-paragraph'>Lorem ipsum dolor sit amet.</p>
+~~~
+~~~css
+:root {
+  --main-color: red;
+}
+
+.cool-paragraph {
+  color: var(--main-color);
+}
+
+.exciting-paragraph {
+  background-color: var(--main-color);
+}
+~~~
+
+By declaring our custom property on the `:root` selector in the example above, we can access it on *any* other valid selector within our CSS file, since any other selector would be considered a descendant of the `:root` selector.
+
+### Themes
+
+Beyond allowing us to access custom properties more globally, the `:root` selector gives us one way to add themes to our pages. Keep in mind that we're not covering every way to add theming in this lesson, as we just want to introduce you to a simple way that you can get a lot of use out of.
+
+~~~html
+<html class='dark'>
+<!-- our boilerplate... -->
+  <body>
+    <div class='themed-element'>Our themed text</div>
+  </body>
+</html>
+~~~
+~~~css
+:root.dark {
+  --color-background: rgb(18, 18, 18);
+  --color-text: rgb(255, 255, 255);
+}
+
+:root.light {
+  --color-background: rgb(255, 255, 255);
+  --color-text: rgb(18, 18, 18);
+}
+
+.themed-element {
+  background-color: var(--color-background);
+  color: var(--color-text);
+}
+~~~
+
+We just created some quick themes! So what is actually going on here? First, we added a `class` attribute on our `html` element in the HTML file so that we can have a more specific selector to add custom properties onto, and so that our page has a default theme.
+
+In our CSS file we're just creating two scopes for our custom properties on the `:root` selector, one for when our `html` (or root) element has a class of `dark`, and another for when it has a class of `light`. Then our `.themed-element` selector will use the values of the custom properties depending on which class is currently present on our root element.
+
+Besides just adding custom properties to the CSS file, we need to add in some JavaScript to handle changing the theme, as well as some interactive element for the user to actually change it (such as a toggle or a switch). At this point, we trust that you can easily find out how to do both of these since that's a little out of scope for this lesson.
```

Comment:
> At this point, we trust that you can easily find out how to do both of these since that's a little out of scope for this lesson.

I don't think we need this part, the reader should understand that.